git-hub

```diff
@@ -1471,6 +1474,143 @@ class PullCmd (IssueCmd):
+                    '(closed at {closed_at})!', **pull)
+
+            remote_url = pull['head']['repo'][config.urltype]
+            remote_branch = pull['head']['ref']
+
+            infof("Fetching '{}' from {}", remote_branch, remote_url)
+            git_quiet(1, 'fetch', remote_url, remote_branch)
+
+            local_branch = 'pull_' + args.pull
+
+            # Warn if the pull branch already exists locally
+            try:
+                pull_branch_hash = git_quiet(1, 'rev-parse', '--short',
+                    '--verify', local_branch)
+                warnf("Pull branch '{}' already exists at commit '{}'. "
+                    "It _may_ get moved.", local_branch, pull_branch_hash)
+                git_quiet(1, 'checkout', '-B', local_branch, 'FETCH_HEAD')
+            except GitError:
+                infof("Checking out locally as branch '{}'", local_branch)
+                git_quiet(1, 'checkout', '-b', local_branch, 'FETCH_HEAD')
+
+            commits_range = args.base + '..' + local_branch
+            commits_list = git('log', '--oneline', commits_range).split('\n')
+            infof("Found {} commits in range '{}'",
+                len(commits_list), commits_range)
+
+            if args.auto_skip:
+                if args.auto_skip >= len(commits_list):
+                    infof('All commits automatically skipped')
+                    return
+                else:
+                    if args.auto_skip == 1:
+                        infof('Skipping commit 1')
+                    elif args.auto_skip == 2:
+                        infof('Skipping commits 1 & 2')
+                    else:
+                        infof('Skipping commits 1 to {}', args.auto_skip)
+
+            commit_num = 0
+            num_commits_auto_skipped = 0
+
+            for c in reversed(commits_list):
+                commit_num += 1
+
+                if args.auto_skip and num_commits_auto_skipped < args.auto_skip:
+                    num_commits_auto_skipped += 1
+                    continue
+
+                current_commit, commit_msg_title = c.split(' ', 1)
+
+                infof('')
+                sys.stdout.write('    ' + str(commit_num) + '. (\033[94m' +
+                    current_commit + '\033[0m) \033[1m' + commit_msg_title +
+                    '\033[0m\n')
+
+                shortstat_list = git('show', '--shortstat', '--format=%b',
+                    current_commit).split('\n')
+
+                # If the shortstat list has less than 3 elements, then the
+                # commit message only has a title, but no body
+                if len(shortstat_list) > 3:
+                    for line in shortstat_list[0:-3]:
+                        infof('            {}', line)
+
+                files_changed_str = shortstat_list[-1]
+                sys.stdout.write('\033[35m' + files_changed_str.rjust(84) +
+                    '\n')
+                sys.stdout.write("press 'r' to review, 's' to skip".rjust(84) +
```

Comment:
There is the `ask()` function for this. Yes, you need to press ENTER, but it's not that bad (and if we were to change that we should do so for all questions to the user).